include/aspect/mesh_refinement/isosurfaces.h

```diff
@@ -0,0 +1,146 @@
+/*
+  Copyright (C) 2014 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef _aspect_mesh_refinement_isosurfaces_h
+#define _aspect_mesh_refinement_isosurfaces_h
+
+#include <aspect/mesh_refinement/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+#include <deal.II/base/parsed_function.h>
+
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    namespace Internal
+    {
+      enum class PropertyType
+      {
+        Temperature,
+        Composition
+      };
+
+      struct Property
```

Comment:
Yes, currently is functions very similar to the `Simulator::AdvectionField` with a few extra convenient functions. I did not realize this and looked into it, but I don't think it can be replaced, because the plan is for the property to also contain other things than advection fields such as density or strain-rate.